Gems/GradientSignal/Code/Source/Editor/EditorImageGradientComponent.cpp

```diff
@@ -563,11 +706,47 @@ namespace GradientSignal
             imageResolutionX, imageResolutionY, channels, m_outputFormat, pixelBuffer);
     }
 
+    AZStd::string EditorImageGradientComponent::GetImageSourcePath(const AZ::Data::AssetId& imageAssetId) const
+    {
+        if (imageAssetId.IsValid())
+        {
+            AZStd::string sourcePath;
+            bool sourceFileFound = false;
+            AZ::Data::AssetInfo assetInfo;
+            AZStd::string watchFolder;
+
+            AzToolsFramework::AssetSystemRequestBus::BroadcastResult(
+                sourceFileFound,
+                &AzToolsFramework::AssetSystem::AssetSystemRequest::GetSourceInfoBySourceUUID,
+                imageAssetId.m_guid,
+                assetInfo,
+                watchFolder);
+
+            if (sourceFileFound)
+            {
+                bool success = AzFramework::StringFunc::Path::ConstructFull(
+                    watchFolder.c_str(), assetInfo.m_relativePath.c_str(), sourcePath, true);
+
+                if (success)
+                {
+                    return sourcePath;
+                }
+            }
+        }
+
+        // Invalid image asset or failed path creation, try creating a new name.
```

Comment:
Should we log a warning if we get to this point?

Reply:
We can end up here through the "Create Image" flow with an intentionally-invalid image asset, so we shouldn't log a warning. I _think_ we already log an error in other cases where we could get here, but I'll examine the code further and see if there's anywhere else that needs warnings.